### PR TITLE
update the server to os.mbed.com

### PR DESCRIPTION
**Pull request by studavekar**
Related :
https://github.com/ARMmbed/mbed-os/pull/5122

### CODE REVIEW

**Comment by andresag01** [comment]
Thanks for the PR. I think the changes look good and I already checked that the server certificate still checks out with the PEM CA in the tls-client program.

Currently just waiting for the CI to give us a pass.

File: tls-client/main.cpp [hunk 1]
```diff
@@ -50,7 +50,7 @@
 
 namespace {
 
-const char *HTTPS_SERVER_NAME = "developer.mbed.org";
+const char *HTTPS_SERVER_NAME = "os.mbed.org";
```
Comment (andresag01) [inline tls-client/main.cpp]: On second thought, it seems that the domain should actually be `os.mbed.com`. I tried `os.mbed.org` and that says the site is unreachable

**Comment by studavekar** [comment]
Corrected typo, Output of test run below.

```
Using Ethernet LWIP
Client IP Address is 10.118.12.134
Connecting with os.mbed.com
Starting the TLS handshake...
TLS connection to os.mbed.com established
Server certificate:
    cert. version     : 3
    serial number     : 65:7B:6D:8D:15:A5:B6:86:87:6B:5E:BC
    issuer name       : C=BE, O=GlobalSign nv-sa, CN=GlobalSign Organization Validation CA - SHA256 - G2
    subject name      : C=GB, ST=Cambridgeshire, L=Cambridge, O=ARM Ltd, CN=*.mbed.com
    issued  on        : 2017-04-03 13:54:02
    expires on        : 2018-05-06 10:31:02
    signed using      : RSA with SHA-256
    RSA key size      : 2048 bits
    basic constraints : CA=false
    subject alt name  : *.mbed.com, mbed.org, *.mbed.org, mbed.com
    key usage         : Digital Signature, Key Encipherment
    ext key usage     : TLS Web Server Authentication, TLS Web Client Authentication
Certificate verification passed
```

File: tls-client/main.cpp (hunk 1 above)
Comment (studavekar) [reply tls-client/main.cpp]: Oops my bad , updated the PR.

**Comment by andresag01** [comment]
@studavekar: Thank you for the fix. I will wait for the CI results.

**Comment by RonEld** [comment]
Hi @studavekar 
The output you showed is also printed when the hostname is developer.mbed.org:
~~~
Using Ethernet LWIP
Client IP Address is 172.16.8.12
Connecting with developer.mbed.org
Starting the TLS handshake...
TLS connection to developer.mbed.org established
Server certificate:
    cert. version     : 3
    serial number     : 65:7B:6D:8D:15:A5:B6:86:87:6B:5E:BC
    issuer name       : C=BE, O=GlobalSign nv-sa, CN=GlobalSign Organization Validation CA - SHA256 - G2
    subject name      : C=GB, ST=Cambridgeshire, L=Cambridge, O=ARM Ltd, CN=*.mbed.com
    issued  on        : 2017-04-03 13:54:02
    issued  on        : 2017-04-03 13:54:02                                                                              
    expires on        : 2018-05-06 10:31:02                                                                              
    signed using      : RSA with SHA-256                                                                                 
    RSA key size      : 2048 bits                                                                                        
    basic constraints : CA=false                                                                                         
    key usage         : Digital Signature, Key Encipherment
    ext key usage     : TLS Web Server Authentication, TLS Web Client Authentication
Certificate verification passed
~~~
When changed to os.mbed.com the output should be:
~~~
Using Ethernet LWIP
Client IP Address is 172.16.8.12
Connecting with os.mbed.com
Starting the TLS handshake...
TLS connection to os.mbed.com established
Server certificate:
    cert. version     : 3
    serial number     : 65:7B:6D:8D:15:A5:B6:86:87:6B:5E:BC
    issuer name       : C=BE, O=GlobalSign nv-sa, CN=GlobalSign Organization Validation CA - SHA256 - G2
    subject name      : C=GB, ST=Cambridgeshire, L=Cambridge, O=ARM Ltd, CN=*.mbed.com
    issued  on        : 2017-04-03 13:54:02
    expires on        : 2018-05-06 10:31:02
    signed using      : RSA with SHA-256
    RSA key size      : 2048 bits
    basic constraints : CA=false
    subject alt name  : *.mbed.com, mbed.org, *.mbed.org, mbed.com
    subject alt name  : *.mbed.com, mbed.org, *.mbed.org, mbed.com                                                       
    key usage         : Digital Signature, Key Encipherment                                                              
    ext key usage     : TLS Web Server Authentication, TLS Web Client Authentication                                     
Certificate verification passed                                                                                          
                                                                                                                         
HTTPS: Received 200 OK status ... [OK]
HTTPS: Received 'Hello world!' status ... [OK]
HTTPS: Received message:

HTTP/1.1 200 OK
Server: nginx/1.11.12
Date: Mon, 18 Sep 2017 12:54:59 GMT
Content-Type: text/plain
Content-Length: 14
Connection: keep-alive
Last-Modified: Fri, 27 Jul 2012 13:30:34 GMT
Accept-Ranges: bytes
Cache-Control: max-age=36000
Expires: Mon, 18 Sep 2017 22:54:59 GMT
X-Upstream-L3: 172.17.0.4:80
X-Upstream-L2: developer-sjc-cyan-1-nginx
Strict-Transport-Security: max-age=31536000; includeSubdomains

Hello world!
~~~
with the HTTPS response
Please confirm

**Comment by andresag01** [comment]
@RonEld: According to the CI logs, the correct HTTPS response with the "Hello world!" message are being received correctly.

**Comment by andresag01** [comment]
Closing this PR as #117 contains these changes and many other references to developer.mbed.org that needed to be updated to os.mbed.com

**Pull request closed without merging**